android/app/src/main/java/app/organicmaps/SplashActivity.java

```diff
@@ -172,6 +172,12 @@ private void init()
   @SuppressWarnings({"unused", "unchecked"})
   public void processNavigation()
   {
+    if (isDestroyed())
```

Comment:
This should be added to almost every JNI to Java call...

Reply:
Why? That doesn't look like a reliable solution.

Reply:
> Why? That doesn't look like a reliable solution.

What is not clear in this situation? JNI can call Java callbacks at any point in the lifecycle. Some JNI callbacks are removed from onDestroy(), but there are lot of callbacks are static. Especially funny are callback called via event loop.

> What causes it to be called in the first place? Should the destruction start properly and deactivate all native subscribers first?

The callback is called after the end of async map initialization. This is a regression from the previous release.